monkey/monkey_island/cc/island_event_handlers/set_agent_configuration_per_island_mode.py

```diff
@@ -0,0 +1,32 @@
+from typing import Any
+
+from common.agent_configuration import AgentConfiguration
+from monkey_island.cc.models import IslandMode
+from monkey_island.cc.repository import IAgentConfigurationRepository
+
+
+class set_agent_configuration_per_island_mode:
+    """
+    Callable class that sets the Island's mode
+    """
+
+    def __init__(
+        self,
+        agent_configuration_repository: IAgentConfigurationRepository,
+        default_agent_configuration: AgentConfiguration,
+        default_ransomware_agent_configuration: AgentConfiguration,
+    ):
+        self._agent_configuration_repository = agent_configuration_repository
+        self._default_agent_configuration = default_agent_configuration
+        self._default_ransomware_agent_configuration = default_ransomware_agent_configuration
+
+    def __call__(self, event: Any = None):
```

Comment:
These are Island events, we can name these parameters whatever we want, as long as they're consistent.

I think we should just name the parameter `mode`.